bestie/server/xml_result.go

```diff
@@ -197,11 +200,20 @@ func parseUnstructuredXml(ts *TestSuite) ([]*xmlResult, error) {
 				matches[key] = val
 				key = ""
 			}
-		} else if strings.Contains(line, " SKIPPEDSESSIONFINISH:") ||
-			strings.Contains(line, " SKIPPED") {
+		} else if strings.Contains(line, " SKIPPEDSESSIONFINISH:") {
 			key = strings.Split(strings.Split(line, " ")[0], "[")[0]
 			matches[key] = "skipped"
 			key = ""
+		} else {
+			s := strings.Split(line, " ")
+			if len(s) == 2 {
+				if strings.Contains("FAILED PASSED SKIPPED", s[1]) {
+					// e.g.: systest/trial/test_app.py::TestHello::test_app_metrics_nofile PASSED
+					key = strings.Split(s[0], "[")[0]
+					matches[key] = strings.ToLower(s[1])
+					key = ""
+				}
+			}
```

Comment:
I found another format being used when --junitxml was not specified, meaning unstructured console output needed to be parsed manually. Pytest printed a summary of each test case and its result, one per line. This is what I've referred to as an abbreviated format, previously seen only with skipped test cases. Turns out, in some situations, this is the only indication of failed/passed/skipped test case results.

Reply:
> when --junitxml was not specified

Let's just explicitly not support this case. This happens when the test binary doesn't write a test.xml, and so bazel takes  the test's stdout and writes its own basic test.xml. I really don't want to be in the business of parsing this - it can't possibly be correct for all test frameworks. Let's just make sure the test frameworks emit structured data if its intended to be parsed.

Reply:
I agree that it's going to be an ongoing pain to parse unstructured console output. My intent was to be able to produce pass/fail metrics for all test apps, even ones that do not have the ability to create the junitxml structured output. It'll be simpler (and less error-prone) to just not support unstructured XML. I'll probably archive this offline for future reference, in case the hardware-related tests cannot provide structured XML results.

Reply:
Removed support for an unstructured test.xml file from the latest commit. Now incrementing a service counter and outputting a debug log message (when --debug flag is specified on the command line) when such .xml files are seen.